daemon/logger/fluentd/fluentd.go

```diff
@@ -0,0 +1,99 @@
+package fluentd
+
+import (
+	"io"
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/fluent/fluent-logger-golang/fluent"
+)
+
+type Fluentd struct {
+	tag           string
+	containerID   string
+	containerName string
+	writer        *fluent.Fluent
+}
+
+const name = "fluentd"
+
+const defaultHostName = "localhost"
+const defaultPort = 24224
+const defaultTagPrefix = "docker"
+
+func init() {
+	if err := logger.RegisterLogDriver(name, New); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func New(ctx logger.Context) (logger.Logger, error) {
+	config := ctx.Config
+	port := defaultPort
+	host := defaultHostName
+	if config["port"] != "" {
+		portp, errp := strconv.Atoi(config["port"])
+		if errp != nil {
+			return nil, errp
+		}
+		port = portp
+	}
+	if config["host"] != "" {
+		host = config["host"]
+	}
+	tag := config["tag"]
```

Comment:
I see that but can't the same be achieved with prefix and type?  ngnix.container_id seems OK isn't it? I'm trying to have a pattern here for what is possible with other log drivers also.. And a totally arbitrary tag IMO will made only confusion when you're looking for a particular ngnix log in a particular container (in any driver)